drivers/hmis/app/controllers/hmis/sessions_controller.rb

```diff
@@ -5,11 +5,16 @@
 ###
 
 class Hmis::SessionsController < Devise::SessionsController
+  include Hmis::Concerns::JsonErrors
   include AuthenticatesWithTwoFactor
   respond_to :json
   skip_before_action :verify_signed_out_user
   before_action :authenticate_with_2fa, only: [:create], if: -> { two_factor_enabled? }
 
+  rescue_from ActionController::InvalidAuthenticityToken do
+    render_json_error(401, :unauthenticated)
+  end
```

Comment:
Because we use `protect_from_forgery with: :exception`, devise raises an `ActionController::InvalidAuthenticityToken` exception when the csrf token is missing or invalid.

Right now, that leads to a 422 response (in development) or a 500 response (in prod - I think). We really want to return a 401 here, though, because the user is effectively not authenticated. That lets us have simpler logic on the frontend: if any request returns a 401, we need to redirect the user to log back in.

Reply:
Maybe we can just `protect_from_forgery with: :some_method_name`  This could be done at the root HMIS level or root GraphQL level if that makes sense.
https://api.rubyonrails.org/classes/ActionController/RequestForgeryProtection/ClassMethods.html
`:reset_session` might give what you're looking for.

Reply:
Thanks @eanders. I think `reset_session` is what we want for the base controller (used for graphql), so I updated that. 
For the sessions controller we need something different, because `reset_session` would allow the user to log in even if the login form does not have a csrf token. I updated the sessions controller to use `handle_unverified_request` instead of `rescue_from`, which should have the same effect but looks cleaner.

Reply:
Excellent!